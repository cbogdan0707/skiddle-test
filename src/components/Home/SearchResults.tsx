import React, { ReactElement } from 'react';
import { EventDetails } from './EventDetails';
import { RawEventsResults } from '../../types/types';

export const SearchResults = ({
  searchResults,
}: {
  searchResults: {
    error: number;
    errormessage: string;
    result: { results: Array<RawEventsResults> };
  };
}): ReactElement => {
  return (
    <>
      {searchResults.error && (
        <div data-testid="fetch-error">Error: {searchResults.errormessage}</div>
      )}
      {searchResults.result &&
        searchResults.result.results &&
        searchResults.result.results.length !== 0 && (
          <section className="skiddle-c-search-results skiddle-u-pt-large">
            <ul className="skiddle-c-search-results-container">
              {searchResults.result.results.map((event: RawEventsResults) => (
                <li key={event.id}>
                  {
                    <EventDetails
                      eventDetails={{
                        id: event.id,
                        imageUrl: event.largeimageurl,
                        eventName: event.eventname,
                        description: event.description,
                        location: event.venue.address,
                        date: event.date,
                      }}
                    />
                  }
                </li>
              ))}
            </ul>
          </section>
        )}
    </>
  );
};
