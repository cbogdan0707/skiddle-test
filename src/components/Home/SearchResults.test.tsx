import React from 'react';
import { render } from '@testing-library/react';
import { SearchResults } from './SearchResults';
import { BrowserRouter } from 'react-router-dom';

const result = {
  results: [
    {
      id: 23232,
      largeimageurl: 'image1',
      eventname: 'event1',
      description: 'description1',
      venue: { address: 'location1' },
      date: 'date1',
    },
    {
      id: 323243,
      largeimageurl: 'image2',
      eventname: 'event2',
      description: 'description2',
      venue: { address: 'location2' },
      date: 'date2',
    },
  ],
};

describe('Event details', () => {
  it('should match snapshot when results are available', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SearchResults searchResults={{ result }} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should display an error message when fetch fails', () => {
    const { queryByTestId } = render(
      <BrowserRouter>
        <SearchResults
          searchResults={{
            error: true,
            results: null,
            errormessage: 'failed to fetch',
          }}
        />
      </BrowserRouter>
    );
    expect(queryByTestId('fetch-error')?.innerHTML).toEqual(
      'Error: failed to fetch'
    );
  });
});
