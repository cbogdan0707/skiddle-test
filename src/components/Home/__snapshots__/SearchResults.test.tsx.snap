// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event details should match snapshot when results are available 1`] = `
<DocumentFragment>
  0
  <section
    class="skiddle-c-search-results skiddle-u-pt-large"
  >
    <ul
      class="skiddle-c-search-results-container"
    >
      <li>
        <div
          class="skiddle-c-event-details"
        >
          <img
            alt="event"
            src="image1"
          />
          <div
            class="skiddle-u-text-align-center skiddle-u-bold skiddle-u-mt-small"
          >
            event1
          </div>
          <div
            class="skiddle-u-mt-small"
          >
            description1
          </div>
          <div
            class="skiddle-u-mt-small skiddle-c-location"
          >
            location1
          </div>
          <div
            class="skiddle-u-mt-small skiddle-c-date"
          >
            date1
          </div>
          <div
            class="skiddle-u-text-align-center skiddle-u-mt-medium"
          >
            <a
              class="skiddle-u-primary-button skiddle-u-pa-small"
              href="/event/23232"
            >
              view details
            </a>
          </div>
        </div>
      </li>
      <li>
        <div
          class="skiddle-c-event-details"
        >
          <img
            alt="event"
            src="image2"
          />
          <div
            class="skiddle-u-text-align-center skiddle-u-bold skiddle-u-mt-small"
          >
            event2
          </div>
          <div
            class="skiddle-u-mt-small"
          >
            description2
          </div>
          <div
            class="skiddle-u-mt-small skiddle-c-location"
          >
            location2
          </div>
          <div
            class="skiddle-u-mt-small skiddle-c-date"
          >
            date2
          </div>
          <div
            class="skiddle-u-text-align-center skiddle-u-mt-medium"
          >
            <a
              class="skiddle-u-primary-button skiddle-u-pa-small"
              href="/event/323243"
            >
              view details
            </a>
          </div>
        </div>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
